js/modules/k6/ws/ws_test.go

```diff
@@ -1288,3 +1291,70 @@ func TestCookieJar(t *testing.T) {
 
 	assertSessionMetricsEmitted(t, metrics.GetBufferedSamples(ts.samples), "", sr("WSBIN_URL/ws-echo-someheader"), statusProtocolSwitch, "")
 }
+
+func TestWSConnectWithThrowErrorOption(t *testing.T) {
+	t.Parallel()
+	tb := httpmultibin.NewHTTPMultiBin(t)
+	root, err := lib.NewGroup("", nil)
+	require.NoError(t, err)
+
+	logHook := &testutils.SimpleLogrusHook{HookedLevels: []logrus.Level{logrus.WarnLevel}}
+	testLog := logrus.New()
+	testLog.AddHook(logHook)
+	testLog.SetOutput(io.Discard)
+
+	rt := goja.New()
+	rt.SetFieldNameMapper(common.FieldNameMapper{})
+	samples := make(chan metrics.SampleContainer, 1000)
+	state := &lib.State{
+		Group:  root,
+		Dialer: tb.Dialer,
+		Options: lib.Options{
+			SystemTags: &metrics.DefaultSystemTagSet,
+		},
+		Samples:        samples,
+		BuiltinMetrics: metrics.RegisterBuiltinMetrics(metrics.NewRegistry()),
+		Tags:           lib.NewTagMap(nil),
+		Logger:         testLog,
+	}
+
+	m := New().NewModuleInstance(&modulestest.VU{
+		CtxField:     context.Background(),
+		InitEnvField: &common.InitEnvironment{},
+		RuntimeField: rt,
+		StateField:   state,
+	})
+	require.NoError(t, rt.Set("ws", m.Exports().Default))
+
+	t.Run("ThrowEnabled", func(t *testing.T) {
+		t.Parallel()
+		state.Options.Throw = null.BoolFrom(true)
+		_, err = rt.RunString(`
+		var res = ws.connect("INVALID", function(socket){
+			socket.on("open", function() {
+				socket.close();
+			});
+		});
+		`)
+		assert.Error(t, err)
+	})
+
+	t.Run("ThrowDisabled", func(t *testing.T) {
+		t.Parallel()
+		state.Options.Throw = null.BoolFrom(false)
```

Comment:
This will `race` with the other setting in the previous test as you are setting the same object.'

I would recommend just using `newTestState` and creating a new one in each of the subtests